plugins/wiggle/src/LinearWiggleDisplay/models/model.ts

```diff
@@ -72,14 +82,37 @@ const stateModelFactory = (configSchema: ReturnType<typeof ConfigSchemaF>) =>
         }
         self.statsFetchInProgress = aborter
       },
+
+      setResolution(res: number) {
+        self.resolution = res
+      },
+
+      setFill(fill: boolean) {
+        self.fill = fill
+      },
+      setLogScale(log: boolean) {
+        self.logScale = log
+      },
+      setAutoscale(val: string) {
+        self.autoscale = val
+      },
     }))
     .views(self => {
+      const { trackMenuItems } = self
       let oldDomain: [number, number] = [0, 0]
       return {
         get TooltipComponent(): React.FC {
           return (Tooltip as unknown) as React.FC
         },
 
+        get adapterTypeName() {
+          let curr = self
+          while (!curr.configuration || !curr.configuration.trackId) {
+            curr = getParent(curr)
+          }
+          return getConf(curr, ['adapter', 'type'])
```

Comment:
FYI There's a shortcut on the BaseDisplayModel for getting an adapter config.

```suggestion
          return self.adapterConfig.type
```

Reply:
ah...good to know